plugins/Compressor/CompressorControlDialog.cpp

```diff
@@ -0,0 +1,753 @@
+	}
+	m_lastKneePoint = kneePoint;
+
+	p.end();
+
+	if (m_controls->m_effect->m_redrawThreshold)
+	{
+		p.begin(&m_miscPixmap);
+
+		p.setCompositionMode(QPainter::CompositionMode_Source);
+		p.fillRect(0, 0, m_windowSizeX, m_windowSizeY, QColor("transparent"));
+		p.setCompositionMode(QPainter::CompositionMode_SourceOver);
+
+		p.setRenderHint(QPainter::Antialiasing, true);
+
+		// Draw threshold lines
+		p.setPen(QPen(m_threshColor, 2, Qt::DotLine));
+		p.drawLine(0, threshYPoint, m_windowSizeX, threshYPoint);
+		p.drawLine(threshXPoint, 0, threshXPoint, m_kneeWindowSizeY);
+
+		p.end();
+
+		m_controls->m_effect->m_redrawThreshold = false;
+	}
+
+	m_lastPoint = yPoint;
+	m_lastGainPoint = yGainPoint;
+
+	update();
+}
+
+
+
+void CompressorControlDialog::paintEvent(QPaintEvent *event)
+{
+	if (isVisible())
+	{
+		QPainter p(this);
+
+		p.setCompositionMode(QPainter::CompositionMode_Source);
+		p.fillRect(0, 0, m_windowSizeX, m_windowSizeY, QColor("transparent"));
+		p.setCompositionMode(QPainter::CompositionMode_SourceOver);
+
+		p.drawPixmap(0, 0, m_graphPixmap);
+		p.drawPixmap(0, 0, m_visPixmap);
+		p.setOpacity(0.25);
+		p.drawPixmap(0, 0, m_kneePixmap);
+		p.setOpacity(1);
+		if (m_controls->m_effect->isEnabled() && m_controls->m_effect->isRunning())
+		{
+			p.drawPixmap(0, 0, m_kneePixmap2);
+		}
+		p.drawPixmap(0, 0, m_miscPixmap);
+
+		p.end();
+	}
+}
+
+
+inline int CompressorControlDialog::dbfsToYPoint(float inDbfs)
+{
+	return (-((inDbfs + m_dbRange) / m_dbRange) + 1) * m_windowSizeY;
+}
+
+inline int CompressorControlDialog::dbfsToXPoint(float inDbfs)
+{
+	return m_kneeWindowSizeY - dbfsToYPoint(inDbfs);
+}
+
+
+void CompressorControlDialog::resizeEvent(QResizeEvent *event)
+{
+	resetCompressorView();
+}
+
+
+void CompressorControlDialog::wheelEvent(QWheelEvent * event)
+{
+	const float temp = m_dbRange;
+	m_dbRange = round(qBound(3.f, m_dbRange - event->delta() / 20.f, 96.f) / 3.f) * 3.f;
+
+	// Only reset view if the scolling had an effect
+	if (m_dbRange != temp)
+	{
+		resetGraph();
+		m_controls->m_effect->m_redrawKnee = true;
+		m_controls->m_effect->m_redrawThreshold = true;
+	}
+}
+
+
+void CompressorControlDialog::resetGraph()
+{
+	QPainter p;
+
+	p.begin(&m_graphPixmap);
+
+	p.setRenderHint(QPainter::Antialiasing, false);
+
+	p.setCompositionMode(QPainter::CompositionMode_Source);
+	p.fillRect(0, 0, m_windowSizeX, m_windowSizeY, QColor("transparent"));
+	p.setCompositionMode(QPainter::CompositionMode_SourceOver);
+
+	p.setPen(QPen(m_textColor, 1));
+	p.setFont(QFont("Arial", qMax(int(m_windowSizeY / 1080.f * 24), 12)));
+
+	// Redraw graph
+	p.setPen(QPen(m_graphColor, 1));
+	for (int i = 1; i < m_dbRange / 3.f + 1; ++i)
+	{
+		p.drawLine(0, dbfsToYPoint(-3 * i), m_windowSizeX, dbfsToYPoint(-3 * i));
+		p.drawLine(dbfsToXPoint(-3 * i), 0, dbfsToXPoint(-3 * i), m_kneeWindowSizeY);
+		p.drawText(QRectF(m_windowSizeX - 50, dbfsToYPoint(-3 * i), 50, 50), Qt::AlignRight | Qt::AlignTop, QString::number(i * -3));
```

Comment:
There's a lot of constants here that could use defining what they mean. I'm seeing `3` a lot.

Reply:
It's 3 db per grid line.  I'll add a const variable for that.